numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -1888,6 +1888,185 @@ zfill_strided_loop(PyArrayMethod_Context *context,
     return -1;
 }
 
+static NPY_CASTING
+string_partition_resolve_descriptors(
+        PyArrayMethodObject *self,
+        PyArray_DTypeMeta *const NPY_UNUSED(dtypes[3]),
+        PyArray_Descr *const given_descrs[3],
+        PyArray_Descr *loop_descrs[3],
+        npy_intp *NPY_UNUSED(view_offset))
+{
+    if (given_descrs[2] || given_descrs[3] || given_descrs[4]) {
```

Comment:
Just looked through the code and we do support output for replace, etc., so I feel we might as well here, just copying its resolution (but perhaps looping over the outputs)...

Reply:
I'd also need to update the ufunc loop to account for possibly in-place operations by adding a path that mallocs the result and then packs it into the output string, which increases the code complexity, all for code that can't work right now.

I'd rather just leave the error here as-is. I wish I hadn't implemented in-place support for `replace`, I was under the mistaken understanding I needed it.

Reply:
OK, that makes sense - it sounds like it would be better to return to it after making some helper functions - e.g., using `NpyString_dup` internally. Added to #25693